### PR TITLE
Add Amp Decorator, amp styles and snapshot tests

### LINKED ISSUE #943: Implement AMP decorator for storybook
**_This is the implementation of an investigation https://github.com/BBC-News/simorgh/issues/911_**

**Is your feature request related to a problem? Please describe.**
When we have an AMP component, e.g. an AMP Image component that renders`<amp-img>`, we want to make it visible in Storybook https://simorghstorybook.now.sh.  
However, because our standard Storybook setup does not include the AMP required markup and JS, (https://www.ampproject.org/docs/fundamentals/spec#required-markup) the component won't render accurately. 

**This is a prerequisite to adding AMP components to our application and also to our future component library.**

**Describe the solution you'd like**
- A storybook setup that displays AMP components
- Implement the [investigation](https://github.com/BBC-News/simorgh/pull/933)  decorator which can be found here https://github.com/BBC-News/simorgh/pull/933/commits/5458dd4b9a7b14efea74d7bf84a4312795aaa8b4

**Testing notes**
Add an AMP component, e.g. an AMP Image component that renders`<amp-img>`, and see it displayed correctly in Storybook https://simorghstorybook.now.sh.  

**Additional context**
N/A

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Resolves #943

Adds storybook decorator to add required markup and JS, (https://www.ampproject.org/docs/fundamentals/spec#required-markup) when displaying an AMP component.

The decorator uses `react-helmet` to add markup that AMP requires including the primary AMP js bundle script. This decorator can then simply be attached to a collection of stories using ```.addDecorator(AmpDecorator)``` 

- [x] Tests added for new features
- [ ] Test engineer approval

Testing - 
Attach the decorator to an existing storybook story and inspect the iframe to see the [expected AMP markup](https://www.ampproject.org/docs/fundamentals/spec#required-markup) present

### CODE REVIEW

**Review by bcmn — DISMISSED** [review]

File: src/app/helpers/storybook/ampDecorator/index.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { ampScript, ampNoscript } from '../../../lib/constants/styles';
+
+/*
+  Contains AMP required markup https://www.ampproject.org/docs/fundamentals/spec#required-markup
+*/
+
+const AmpDecorator = storyFn => (
```
Comment (jtart) [inline src/app/helpers/storybook/ampDecorator/index.jsx:9]: We should pass the components through as `children`

```suggestion
const AmpDecorator = children => (
```

File: src/app/helpers/storybook/ampDecorator/index.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  <div>
+    <Helmet htmlAttributes={{ amp: '' }}>
+      <link rel="canonical" href="http://foobar.com" />
+      <meta charset="utf-8" />
+      <meta name="viewport" content="width=device-width,minimum-scale=1" />
+      <style amp-boilerplate="">{ampScript}</style>
+      <noscript>{`<style amp-boilerplate="">${ampNoscript}</style>`}</noscript>
+      <script async src="https://cdn.ampproject.org/v0.js" />
+    </Helmet>
+    {storyFn()}
```
Comment (jtart) [inline src/app/helpers/storybook/ampDecorator/index.jsx:19]:
```suggestion
    {children}
```

File: src/app/helpers/storybook/ampDecorator/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import AmpDecorator from './index';
+
+describe('AmpDecorator', () => {
+  it('should render correctly', () => {
+    const component = AmpDecorator(() => <div>Foobar</div>);
+    expect(component.props.children).toMatchSnapshot();
```
Comment (jtart) [inline src/app/helpers/storybook/ampDecorator/index.test.jsx]: Can we snapshot the whole component, not just children? Also, there is a [snapshot utility that we should use](https://github.com/BBC-News/simorgh/blob/latest/src/app/helpers/tests/testHelpers.js#L5-L10)

**Review by jtart — CHANGES_REQUESTED** [review]

File: src/app/helpers/storybook/ampDecorator/index.test.jsx (hunk 3 above)
Comment (dr3) [reply src/app/helpers/storybook/ampDecorator/index.test.jsx]: I wasn't able to use this the utility as Helmet stuff doesn't show, so you just get `<div><div> Foobar </div></div>`, I have have changed to snapshotting the whole component though

File: src/app/helpers/storybook/ampDecorator/index.jsx (hunk 2 above)
Comment (dr3) [reply src/app/helpers/storybook/ampDecorator/index.jsx:19]: This needs to be an explicit function call as seen https://storybook.js.org/addons/introduction/#storybook-decorators, which is annoying

File: src/app/helpers/storybook/ampDecorator/index.jsx (hunk 1 above)
Comment (dr3) [reply src/app/helpers/storybook/ampDecorator/index.jsx:9]: As per the other comment I think its misleading for it to be children as thats a react-y convention

**Review by bcmn — APPROVED** [review]

**Review by jtart — APPROVED** [review]

**Pull request merged**